File: app/src/main/java/com/itmoldova/adapter/CategoriesFragmentPagerAdapter.java
package com.itmoldova.adapter;

import android.content.res.Resources;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import com.itmoldova.list.ArticlesFragment;
import com.itmoldova.model.Category;

public class CategoriesFragmentPagerAdapter extends FragmentPagerAdapter {

    private Category[] categories = Category.values();
    private Resources resources;

    public CategoriesFragmentPagerAdapter(FragmentManager fm, Resources resources) {
        super(fm);
        this.resources = resources;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return resources.getString(categories[position].getStringResId());
    }

    @Override
    public Fragment getItem(int position) {
        return ArticlesFragment.newInstance(categories[position]);
    }

    @Override
    public int getCount() {
        return categories.length;
    }
}

File: app/src/main/java/com/itmoldova/adapter/CategoriesFragmentPagerAdapter.kt
package com.itmoldova.adapter

import android.content.res.Resources
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

import com.itmoldova.list.ArticlesFragment
import com.itmoldova.model.Category

class CategoriesFragmentPagerAdapter(fm: FragmentManager, private val resources: Resources) : FragmentPagerAdapter(fm) {

    private val categories = Category.values()

    override fun getPageTitle(position: Int): CharSequence =
            resources.getString(categories[position].stringResId)

    override fun getItem(position: Int): Fragment = ArticlesFragment.newInstance(categories[position])

    override fun getCount(): Int = categories.size
}